aces/Loris
# [instruments] Add instrument data dictionary to module

This extracts the part of PR#6936 related to instruments'
data dictionary into a new PR for reviewability.

LINST instruments' dictionaries are creating while parsing the
instrument file.

PHP instruments may use the trait LorisFormDictionaryImpl
in order to implement the data dictionary using the same
mechanism that lorisform_parser.php uses to populate the
parameter_type table (or may hardcore the dictionary themselves
in order to avoid the overhead of parsing LorisForm.) The reason
this is done with a trait instead of being implemented directly
in the NDB_BVL_Instrument class is that there's no reason to assume
for new instruments that the dictionary is generated by the form.
It's just as reasonable to write a new instrument by doing the opposite:
starting with the dictionary and generating the form based on the
dictionary.

# Testing Instructions
1. Use the LorisFormDictionaryImpl in a variety of PHP instruments
2. Print the data dictionary from the getDataDictionary method and ensure that it looks correct
3. Do the same for LINST instruments with a variety of field types.
